### PR TITLE
[ADD] mail_template_attach_existing_attachment

### PR DESCRIPTION
**Pull request by petrus-v**
This module, based on `mail_attach_existing_attachment` allows to define
[REGular EXpression](https://docs.python.org/3/library/re.html) 
to pre-select existing attachment that matched with file name to send in
compose mail.

This module is designed to work in mass mailing to select proper
files to attached from mail template configuration.

### CODE REVIEW

File: mail_template_attach_existing_attachment/models/mail_template.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class MailTemplate(models.Model):
+    "Templates for sending email"
+    _inherit = "mail.template"
+
+    attach_exist_document_regex = fields.Char(
+        string="attachment name pattern (regex)",
```
Comment (hparfr) [inline mail_template_attach_existing_attachment/models/mail_template.py]:
```suggestion
        string="Attachment name pattern (regex)",
```

**Review by hparfr — APPROVED** [review]
Functionnal review ok

**Comment by petrus-v** [comment]
I've missed something in my last rebase... I'll manage it :fearful:

File: mail_template_attach_existing_attachment/wizard/mail_compose_message.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import re
+
+from odoo import api, models, tools
+
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = "mail.compose.message"
+
+    @api.model
+    def _match_attachement(self, re_pattern, model, res_ids):
```
Comment (damdam-s) [inline mail_template_attach_existing_attachment/wizard/mail_compose_message.py:12]:
```suggestion
    def _match_attachment(self, re_pattern, model, res_ids):
```
nitpicking 😄

**Pull request closed without merging**